lib/buildingsync/makers/workflow_maker_phase_zero.rb

```diff
@@ -0,0 +1,917 @@
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# (4) Other than as required in clauses (1) and (2), distributions in any form
+# of modifications or other derivative works may not use the "OpenStudio"
+# trademark, "OS", "os", or any other confusingly similar designation without
+# specific prior written permission from Alliance for Sustainable Energy, LLC.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+require_relative '../workflow_maker'
+module BuildingSync
+  # base class for objects that will configure workflows based on building sync files
+  class PhaseZeroWorkflowMaker < WorkflowMaker
+    def initialize(doc, ns)
+      super
+
+      # load the workflow
+      @workflow = nil
+
+      # log failed scenarios
+      @failed_scenarios = []
+
+      # select base osw for standalone, small office, medium office
+      base_osw = 'phase_zero_base.osw'
+
+      workflow_path = File.join(File.dirname(__FILE__), base_osw)
+      raise "File '#{workflow_path}' does not exist" unless File.exist?(workflow_path)
+
+      File.open(workflow_path, 'r') do |file|
+        @workflow = JSON.parse(file.read)
+      end
+
+      @facility = {}
+      @subsections = []
+
+      # configure the workflow based on properties in the xml
+      configureForDoc(@workflow)
+    end
+
+    def configureForDoc(osw)
+
+      # get the facility floor areas
+      @facility['gross_floor_area'] = nil
+      @facility['heated_and_cooled_floor_area'] = nil
+      @facility['footprint_floor_area'] = nil
+      @doc.elements.each("/#{@ns}:BuildingSync/#{@ns}:Facilities/#{@ns}:Facility/#{@ns}:Sites/#{@ns}:Site/#{@ns}:Buildings/#{@ns}:Building/#{@ns}:FloorAreas/#{@ns}:FloorArea") do |floor_area_element|
+        floor_area = floor_area_element.elements["#{@ns}:FloorAreaValue"].text.to_f
+        next if floor_area.nil?
+
+        floor_area_type = floor_area_element.elements["#{@ns}:FloorAreaType"].text
```

Comment:
This seems like it belongs in ModelMaker.  WorkflowMaker can configure an OSW which includes a Measure that calls ModelMaker.  WorkflowMaker should not duplicate functionality of ModelMaker.